src/main/java/com/fluxninja/aperture/armeria/DecoratingHTTPClient.java

```diff
@@ -0,0 +1,69 @@
+package com.fluxninja.aperture.armeria;
+
+import com.fluxninja.aperture.sdk.ApertureSDK;
+import com.fluxninja.aperture.sdk.ApertureSDKException;
+import com.fluxninja.aperture.sdk.Flow;
+import com.fluxninja.aperture.sdk.FlowStatus;
+import com.linecorp.armeria.client.ClientRequestContext;
+import com.linecorp.armeria.client.HttpClient;
+import com.linecorp.armeria.client.SimpleDecoratingHttpClient;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class DecoratingHTTPClient extends SimpleDecoratingHttpClient {
+    private ApertureSDK apertureSDK;
+    private String featureName;
+
+    public DecoratingHTTPClient(HttpClient delegate, ApertureSDK apertureSDK, String featureName) {
+        super(delegate);
+        try {
+            this.apertureSDK = apertureSDK;
+            this.featureName = featureName;
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) throws Exception {
+        Map<String, String> labels = labelsFromRequest(req);
+        Flow flow = this.apertureSDK.startFlow(featureName, labels);
+
+        if (flow.accepted()) {
+            HttpResponse res;
+            try {
+                res = unwrap().execute(ctx, req);
+                flow.end(FlowStatus.OK);
+            } catch (InterruptedException | ApertureSDKException e) {
```

Comment:
What is InterruptedException?
When does ApertureSDKException happen?

Reply:
Not sure what was the reasoning behind InterruptedException, removed it for now;
ApertureSDKException in this case can happen when trying to end a flow that was already ended.